### PR TITLE
[BugFix] Support for CHARACTER reserved words syntax

### LINKED ISSUE #17480: Improve mysql workbench compatibility
### Steps to reproduce the behavior (Required)
1. show character set where charset = 'utf8mb4'
2. SET CHARACTER SET utf8

### Expected behavior (Required)
Success.
### Real behavior (Required)
You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'character' at line 1
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
3.5.0

### PR DESCRIPTION
**Pull request by Astralidea**
Signed-off-by: Astralidea <astralidea@163.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #17480

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4 [hunk 1]
```diff
@@ -57,6 +57,7 @@ CATALOGS: 'CATALOGS';
 CEIL: 'CEIL';
 CHAIN: 'CHAIN';
 CHAR: 'CHAR';
+CHARACTER: 'CHARACTER';
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:60]: Should add this to nonReserve in StarRocks.g4

Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:60]: I think is no, in this doc https://dev.mysql.com/doc/refman/5.7/en/keywords.html
CHARACTER (R) is Reserved Words.

**Review by imay — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**